presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/jmx/StatisticsAwareJdbcClient.java

```diff
@@ -72,162 +72,162 @@ public JdbcClientStats getStats()
     @Override
     public boolean schemaExists(JdbcIdentity identity, String schema)
     {
-        return stats.schemaExists.wrap(() -> getDelegate().schemaExists(identity, schema));
+        return stats.schemaExists.wrap(() -> super.schemaExists(identity, schema));
```

Comment:
Maybe we should make `TransactionScopeCachingJdbcClient` to be consistent with `StatisticsAwareJdbcClient`?  I prefer using `getDelegate()` over `super`

Reply:
super is shorter :)
yeah.. no strong opinion

Reply:
i think i slightly prefer `super`. has nicer colors, so reads better.

Reply:
When I see `super` I always need to think twice and I am checking the parent then (which is a bit crazy). While with `getDelegate()` I am within my comfort zone. Quiet and warm.

Reply:
I see where you're coming from.
On the other hand, I'm overriding only selected methods. All other ones are as-if i was always calling `super`. So `super` guarantees me the default behavior.

Reply:
Notice that https://github.com/prestosql/presto/pull/2290 is removing `TransactionScopeCachingJdbcClient`, so following this class is not much an argument to me.

I understand we could different point of views. I think I can get used to use either way. However, I would like to be consistent across the project. I used to use getter method in all `Forwarding*` classes I wrote. See for example io.prestosql.security.ForwardingAccessControl and io.prestosql.plugin.base.security.ForwardingSystemAccessControl. I don't want us go back and forth.

Reply:
I dropped the contentious commit.